stratum/src/lib.rs

```diff
@@ -83,18 +87,37 @@ impl Stratum {
 			dispatcher: dispatcher,
 			workers: Arc::new(RwLock::new(HashMap::new())),
 			secret: secret,
+			notify_counter: RwLock::new(NOTIFY_CONTER_INITIAL),
 		});
 
 		let mut delegate = IoDelegate::<Stratum>::new(stratum.clone());
-		delegate.add_method("miner.subscribe", Stratum::subscribe);
-		delegate.add_method("miner.authorize", Stratum::authorize);
+		delegate.add_method("mining.subscribe", Stratum::subscribe);
+		delegate.add_method("mining.authorize", Stratum::authorize);
+		delegate.add_method("mining.submit", Stratum::submit);
 		stratum.handler.add_delegate(delegate);
 
 		try!(stratum.rpc_server.run_async());
 
 		Ok(stratum)
 	}
 
+	fn submit(&self, params: Params) -> std::result::Result<jsonrpc_core::Value, jsonrpc_core::Error> {
+		Ok(match params {
```

Comment:
because this part is supposed to be shared for different clients (non-ethereum also),
it does not have predefined format, it just passes data upstream removing service messages